addons/stock/models/product.py

```diff
@@ -287,7 +287,7 @@ def _search_qty_available(self, operator, value):
         # to use the quants, not the stock moves. Therefore, we bypass the usual
         # '_search_product_quantity' method and call '_search_qty_available_new' instead. This
         # allows better performances.
-        if value == 0.0 and operator == '>' and not ({'from_date', 'to_date'} & set(self.env.context.keys())):
+        if (not ({'from_date', 'to_date'} & set(self.env.context.keys()))):
```

Comment:
@nim-odoo do we really need restrict it for condition ``qty_available > 0``?
https://github.com/odoo/odoo/commit/9daf3f85a8a3b012a79226fe13d3cde2293fcce0

Reply:
I think the purpose of the original commit was to target the very specific case where we would need to know if there was some quantity available (= we are sure quants exists). I guess I set the condition on the value and the operator to be the least disruptive possible.

I'd say if you are searching for a product without stock available (so there might be not quants), your patch may cause issues.

Reply:
It's not clear for me why it could give different results with not optimized search. For me, not optimized search without date filters does exactly the same as ``_search_qty_available_new``

https://github.com/odoo/odoo/blob/9edf5abba90f1c2bff898b85ade4f0e93daa9ad5/addons/stock/models/product.py#L332-L346

Not optimized search:
https://github.com/odoo/odoo/blob/9edf5abba90f1c2bff898b85ade4f0e93daa9ad5/addons/stock/models/product.py#L124-L125
https://github.com/odoo/odoo/blob/9edf5abba90f1c2bff898b85ade4f0e93daa9ad5/addons/stock/models/product.py#L134-L141
https://github.com/odoo/odoo/blob/9edf5abba90f1c2bff898b85ade4f0e93daa9ad5/addons/stock/models/product.py#L160
https://github.com/odoo/odoo/blob/9edf5abba90f1c2bff898b85ade4f0e93daa9ad5/addons/stock/models/product.py#L182

Reply:
@amoyaux ?

Reply:
Seems correct even if the whole file should be improved on master...

Reply:
I thought to apply this in v14 after accepting for master. Plus refactoring of the file could throw me far from performance matters.
Could we merge this variant? Or maybe pass it through RD flow and merge. So, I would make a PR to v14 then and stock team would make a refactoring in master?

Reply:
@yelizariev Yeah, I was not expecting you to do it. I was more speaking for a member of the logistic team.

Reply:
@amoyaux so what do you think about merging it in v14? https://github.com/odoo/odoo/pull/66652

Reply:
Ok. No problem for me